mantle/kola/tests/misc/boot-mirror-luks.go

```diff
@@ -0,0 +1,234 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package misc
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/coreos/mantle/kola"
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/kola/tests/util"
+	"github.com/coreos/mantle/platform"
+	"github.com/coreos/mantle/platform/conf"
+	"github.com/coreos/mantle/platform/machine/unprivqemu"
+	"github.com/coreos/mantle/system"
+)
+
+var (
+	bootmirrorluks = []byte(`{
+		"ignition": {
+		  "version": "3.2.0"
+		},
+		"storage": {
+		  "disks": [
+			{
+			  "device": "/dev/vda",
+			  "partitions": [
+				{
+				  "label": "bios-1",
+				  "sizeMiB": 1,
+				  "typeGuid": "21686148-6449-6E6F-744E-656564454649"
+				},
+				{
+				  "label": "esp-1",
+				  "sizeMiB": 127,
+				  "typeGuid": "C12A7328-F81F-11D2-BA4B-00A0C93EC93B"
+				},
+				{
+				  "label": "boot-1",
+				  "sizeMiB": 384
+				},
+				{
+				  "label": "root-1"
+				}
+			  ],
+			  "wipeTable": true
+			},
+			{
+			  "device": "/dev/vdb",
+			  "partitions": [
+				{
+				  "label": "bios-2",
+				  "sizeMiB": 1,
+				  "typeGuid": "21686148-6449-6E6F-744E-656564454649"
+				},
+				{
+				  "label": "esp-2",
+				  "sizeMiB": 127,
+				  "typeGuid": "C12A7328-F81F-11D2-BA4B-00A0C93EC93B"
+				},
+				{
+				  "label": "boot-2",
+				  "sizeMiB": 384
+				},
+				{
+				  "label": "root-2"
+				}
+			  ],
+			  "wipeTable": true
+			}
+		  ],
+		  "filesystems": [
+			{
+			  "device": "/dev/md/md-esp",
+			  "format": "vfat",
+			  "label": "EFI-SYSTEM",
+			  "wipeFilesystem": true
+			},
+			{
+			  "device": "/dev/md/md-boot",
+			  "format": "ext4",
+			  "label": "boot",
+			  "wipeFilesystem": true
+			},
+			{
+			  "device": "/dev/mapper/root",
+			  "format": "xfs",
+			  "label": "root",
+			  "wipeFilesystem": true
+			}
+		  ],
+		  "luks": [
+			{
+			  "clevis": {
+				"tpm2": %v
+			  },
+			  "device": "/dev/md/md-root",
+			  "label": "luks-root",
+			  "name": "root",
+			  "wipeVolume": true
+			}
+		  ],
+		  "raid": [
+			{
+			  "devices": [
+				"/dev/disk/by-partlabel/esp-1",
+				"/dev/disk/by-partlabel/esp-2"
+			  ],
+			  "level": "raid1",
+			  "name": "md-esp",
+			  "options": [
+				"--metadata=1.0"
+			  ]
+			},
+			{
+			  "devices": [
+				"/dev/disk/by-partlabel/boot-1",
+				"/dev/disk/by-partlabel/boot-2"
+			  ],
+			  "level": "raid1",
+			  "name": "md-boot",
+			  "options": [
+				"--metadata=1.0"
+			  ]
+			},
+			{
+			  "devices": [
+				"/dev/disk/by-partlabel/root-1",
+				"/dev/disk/by-partlabel/root-2"
+			  ],
+			  "level": "raid1",
+			  "name": "md-root"
+			}
+		  ]
+		}
+	  }`)
+)
+
+func init() {
+	register.RegisterTest(&register.Test{
+		Run:                  runBootMirrorLUKSTest,
+		ClusterSize:          0,
+		Name:                 `boot-mirror-luks`,
```

Comment:
These names are usually dotted paths in a hierarchy.  Let's find an appropriate place in the hierarchy to put these tests.